conn_http.go

```diff
@@ -30,102 +35,178 @@ import (
 )
 
 func dialHttp(ctx context.Context, addr string, num int, opt *Options) (*httpConnect, error) {
-	url := &url.URL{
-		Scheme: "http",
+	u := &url.URL{
+		Scheme: opt.Scheme,
 		Host:   addr,
 	}
 
-	if opt.TLS != nil {
-		url.Scheme = "https"
+	query := u.Query()
+	query.Set("default_format", "Native")
+	u.RawQuery = query.Encode()
+
+	t := &http.Transport{
+		DialContext: (&net.Dialer{
+			Timeout:   opt.DialTimeout,
+			KeepAlive: opt.ConnMaxLifetime,
+		}).DialContext,
+		MaxIdleConns:          1,
+		IdleConnTimeout:       opt.ConnMaxLifetime,
+		ResponseHeaderTimeout: opt.ReadTimeout,
+		TLSClientConfig:       opt.TLS,
 	}
 
-	connect := &httpConnect{
+	conn := &httpConnect{
 		opt: opt,
-		transport: &http.Transport{
-			DialContext: (&net.Dialer{
-				Timeout:   opt.DialTimeout,
-				KeepAlive: opt.ConnMaxLifetime,
-			}).DialContext,
-			MaxIdleConns:          1,
-			IdleConnTimeout:       opt.ConnMaxLifetime,
-			ResponseHeaderTimeout: opt.ReadTimeout,
-			TLSClientConfig:       opt.TLS,
+		client: &http.Client{
+			Transport: t,
 		},
-		url:      url,
-		location: time.UTC, // TODO: make configurable
+		url: u,
 	}
 
-	if err := connect.ping(ctx); err != nil {
+	rows, err := conn.query(ctx, func(*connect, error) {}, "SELECT timeZone()")
+	if err != nil {
 		return nil, err
 	}
 
-	return connect, nil
+	for rows.Next() {
+		var serverLocation string
+		rows.Scan(&serverLocation)
+		location, err := time.LoadLocation(serverLocation)
+		if err != nil {
+			return nil, err
+		}
+		conn.location = location
+	}
+
+	return conn, nil
 }
 
 type httpConnect struct {
-	opt       *Options
-	url       *url.URL
-	transport *http.Transport
-	location  *time.Location
+	opt      *Options
+	url      *url.URL
+	client   *http.Client
+	location *time.Location
+}
+
+func (h *httpConnect) isBad() bool {
+	if h.client == nil {
+		return true
+	}
+	return false
+}
+
+func writeData(encoder *binary.Encoder, block *proto.Block) error {
+	return block.Encode(encoder, 0)
+}
+
+func readData(decoder *binary.Decoder) (*proto.Block, error) {
+	var block proto.Block
+	if err := block.Decode(decoder, 0); err != nil {
+		return nil, err
+	}
+	return &block, nil
+}
+
+func (h *httpConnect) asyncInsert(ctx context.Context, query string, wait bool) error {
```

Comment:
why not? its supported

Reply:
Not supported via HTTP for now, options should be added to URL in query part. I plan to add it in next PR.

Reply:
:+1: